### ISSUE TITLE
Export resources

### ISSUE THREAD

**Issue by robhunt3r** [marvinroger/Dofus-Tools#1]
I'm trying to use your project, but I can't find a way to export sprites into png or jpg files... I think the code leaks of methods like splitting the resource into path/resource (as names are 0/something) and then getting thoose resources...

If there's a way, I would like to know, thanks!

**Comment by marvinroger** [comment]
Yes there is a way, fortunately, as this is the whole point. If you look at the [D2P decompilation sample](https://github.com/marvinroger/Dofus-Tools#decompilation), you will see this code:

``` python
from pydofus.d2p import D2PReader, InvalidD2PFile

stream = open("./samples/sample.d2p", "rb")  # Open the D2P file in binary mode
try:
    D2P = D2PFile(D2P_stream, False)  # Init the D2P object with a D2P file. Must be a stream (Init = get D2P informations, if second parameter is True, load() is called auto)
    D2P.load()  # Populate the D2P object with the actual files in the above given D2P stream. (Load = load files in the D2P in RAM)
    for name, specs in D2P.files.items():
        pass  # Do whatever you want with name the name of the file and specs, which is {position: {offset: <int>, length: <int>}, (if loaded)binary: ByteArray}
except D2PInvalidFile:  # Raised when the D2P file is incorrect
    pass
```

Look at the code in the try block, there is a for iterating through the files dict.
You can see in the comment after the pass:

`# Do whatever you want with name the name of the file and specs, which is {position: {offset: <int>, length: <int>}, (if loaded)binary: ByteArray}`

In other words, the dict looks like so:

``` python
files["filename.jpg"] = { position: { offset: <the offset of the file in the D2P>, length: <the length of the file in the D2P> }, binary: <the ByteArray representing the file> }
```

You can then write the ByteArray to the file you want.

Hope that was clear, it is an old project and I agree there is no clear docs. ;)

**Comment by robhunt3r** [comment]
Never mind, I've done.

For thoose who wants, code:

```
from pydofus.d2p import D2PReader, InvalidD2PFile
from PIL import Image
from cStringIO import StringIO

stream = open("./content/gfx/maps/worldmap0.d2p", "rb")  # Open the D2P file in binary mode
try:
    D2P = D2PReader(stream, True)  # Init the D2P object with a D2P file. Must be a stream (Init = get D2P informations, if second parameter is True, load() is called auto)
    i=0
    for name, specs in D2P.files.items():
        buff = StringIO(specs['binary'])
        path = name.split("/") # Split everything

        image = Image.open(buff)
        image.save("out/%s" % path[-1]) # -1 means SOMETHING.jpg
        i+=1
except InvalidD2PFile:  # Raised when the D2P file is incorrect
    pass
```

**Comment by marvinroger** [comment]
Oh, I didn't understand what you wanted. Glad you found how to do.

**Comment by robhunt3r** [comment]
@marvinroger Yeah I wrote the issue fast I didn't explained very well. I'm trying to find where are the tiles now (terrain tiles) because I can't find them >.<

Thanks anyway !
